rct229/rules/section5/section5rule28.py

```diff
@@ -0,0 +1,139 @@
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as OST,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.match_lists import match_lists_by_id
+from rct229.utils.std_comparisons import std_equal
+
+
+class Section5Rule28(RuleDefinitionListIndexedBase):
+    """Rule 5 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule28, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule28.BuildingRule(),
+            index_rmr="baseline",
+            id="5-17",
+            description="Opaque surfaces that are not regulated (not part of opaque building envelope) must be modeled the same in the baseline as in the proposed design. ",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule28.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section5Rule28.BuildingRule.UnregulatedSurfaceRule(),
+                index_rmr="baseline",
+            )
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            # Merge into the existing data dict
+            return {
+                **data,
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
```

Comment:
the convention is adding a `_b` to indicate this is baseline and `_p` for proposed.
scc should be fine as long as the key stays in data. For output (calc_val), I prefer to have the full name spelled out.
So we will use `scc_dict_b` instead of `surface_conditioning_category_dict`

Reply:
Yes that sounds very good.

Reply:
Yes, I think we should use `_b` and `_p` in the calc_vals.

At one point, I think we will need to clean up all the existing rules in one PR.